pkg/printers/internalversion/printers.go

```diff
@@ -767,6 +770,13 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 		reason = pod.Status.Reason
 	}
 
+	// If the Pod carries {type:PodScheduled, reason:WaitingForGates}, set reason to SchedulingPaused.
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == api.PodScheduled && condition.Reason == api.PodReasonWaitingForGates {
+			reason = schedulingPaused
```

Comment:
why not use the same reason?

Reply:
It's b/c this "reason" is used in the output of `k get pod`, and thus`PodReasonWaitingForGates` which holds the literal "WaitingForGates" loses the context - a word describing that it's scheduling gated would be more self-descriptive.